internal/pkg/manifest/lb_web_svc.go

```diff
@@ -250,23 +255,42 @@ func (c *NetworkLoadBalancerConfiguration) IsEmpty() bool {
 func (lbws *LoadBalancedWebService) ExposedPorts() (ExposedPortsIndex, error) {
 	var exposedPorts []ExposedPort
 	workloadName := aws.StringValue(lbws.Name)
+	// port from image.port.
 	exposedPorts = append(exposedPorts, lbws.ImageConfig.exposedPorts(workloadName)...)
+	// port from sidecar[x].image.port.
 	for name, sidecar := range lbws.Sidecars {
 		out, err := sidecar.exposedPorts(name)
 		if err != nil {
 			return ExposedPortsIndex{}, err
 		}
 		exposedPorts = append(exposedPorts, out...)
 	}
+	// port from http.target_port.
 	exposedPorts = append(exposedPorts, lbws.RoutingRule.exposedPorts(exposedPorts, workloadName)...)
-	out, err := lbws.NLBConfig.exposedPorts(exposedPorts, workloadName)
-	if err != nil {
-		return ExposedPortsIndex{}, err
+
+	nlbRoutingRules := lbws.NLBConfig.NLBRoutingRules()
+	// port from nlb.target_port and nlb.additional_rules[x].target_port
+	for _, rule := range nlbRoutingRules {
+		out, err := rule.exposedPorts(exposedPorts, workloadName)
+		if err != nil {
+			return ExposedPortsIndex{}, err
+		}
+		exposedPorts = append(exposedPorts, out...)
 	}
-	exposedPorts = append(exposedPorts, out...)
 	portsForContainer, containerForPort := prepareParsedExposedPortsMap(sortExposedPorts(exposedPorts))
 	return ExposedPortsIndex{
 		PortsForContainer: portsForContainer,
 		ContainerForPort:  containerForPort,
 	}, nil
 }
+
+// NLBRoutingRules returns primary as well as additional rules as a list of NetworkLoadBalancerRoutingRule.
+func (cfg NetworkLoadBalancerConfiguration) NLBRoutingRules() []NetworkLoadBalancerRoutingRule {
```

Comment:
Can we move the function close to the definition of its receiver?

Reply:
Is there a unit test for this exposed method?